src/normalize.js

```diff
@@ -289,9 +369,44 @@ const extractImages = async (item, ctx, uid) => {
     //   return extractImages(value, ctx, attribute.target);
     // }
 
-    // Always extract images for components
-
     if (value && type) {
+      if (type === 'richtext') {
+        const extractedFiles = extractFiles(value[attributeName], apiURL);
+
+        const files = await Promise.all(
+          extractedFiles.map(async ({ url }) => {
+            const filters = qs.stringify(
+              {
+                filters: { url: url.replace(`${apiURL}`, '') },
+              },
+              { encode: false }
+            );
+
+            const { data } = await axiosInstance.get(`/api/upload/files?${filters}`);
+            const file = data[0];
+
+            if (!file) {
+              return null;
+            }
+
+            const fileNodeID = await downloadFile(file, ctx);
+
+            return fileNodeID;
+          })
+        );
+
+        const nodeIds = files.filter(Boolean);
+
+        for (let i = 0; i < nodeIds.length; i++) {
+          item[attributeName][`localFiles`].push({
+            alternativeText: extractedFiles[i].alternativeText,
+            url: extractedFiles[i].url,
+            src: extractedFiles[i].src,
+            localFile___NODE: nodeIds[i],
+          });
```

Comment:
what about metadata fulfilled inside strapi panel?

![image](https://user-images.githubusercontent.com/3448993/151442731-649bd5f9-d189-4f85-81c9-eec004ce8039.png)

With this approach, you will miss many critical data. On line 385 you download them, so probably you can desctruct them here.

Reply:
I thought about it and actually when using the richtext the alternativeText you want to use is the one set in your markdown and not necessary the one from the file here's what I did:

```
      content {
        data {
          childMarkdownRemark {
            html
          }
        }
        # Extracted files from the richtext field
        medias {
          localFile {
            childImageSharp {
              gatsbyImageData
            }
          }
          alternativeText
          # Original url in the markdown
          src
          # Prefixed url
          url
          # Original media from the media library
          file
        }
      }
```

Reply:
For example in my case, that is not enough in case that I use also the `caption` field for copyright information.

Reply:
@rroslaniec solved!

You'll get access to the original file under the file key